core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -45,31 +47,66 @@ protected FileScanTask self() {
 
   @Override
   protected FileScanTask newSplitTask(FileScanTask parentTask, long offset, long length) {
-    return new SplitScanTask(offset, length, parentTask);
+    return new SplitScanTask(offset, length, parentTask, deletesSizeBytes());
   }
 
   @Override
   public List<DeleteFile> deletes() {
-    return ImmutableList.copyOf(deletes);
```

Comment:
Just noticed that in the Iceberg version used in Trino, the deletes() does copy on every invocation. Thank you for fixing this!